wremnants/include/theoryTools.h

```diff
@@ -198,6 +200,30 @@ using helicity_scale_tensor_t = Eigen::TensorFixedSize<double, Eigen::Sizes<NHEL
     return original_weight * scale_tensor.reshape(reshapescale).broadcast(broadcasthelicities) * moments.broadcast(broadcastscales);
   }
 
+  helicity_helicity_tensor_t makeHelicityMomentHelicityTensor(const CSVars &csvars, const helicity_tensor_t &hel_tensor, double original_weight = 1.0)
+  {
+
+    constexpr Eigen::Index nhelicity = NHELICITY;
+
+    constexpr std::array<Eigen::Index, 2> broadcasthels = {nhelicity,1};
+    constexpr std::array<Eigen::Index, 2> broadcasthelicities = {1, nhelicity};
+    constexpr std::array<Eigen::Index, 2> reshapehelicities = {1, nhelicity};
+
+    Eigen::TensorFixedSize<double, Eigen::Sizes<nhelicity, 1>> moments = csAngularMoments(csvars).reshape(broadcasthels);
+    helicity_helicity_tensor_t hel_hel_tensor = original_weight * hel_tensor.reshape(reshapehelicities).broadcast(broadcasthels) * moments.broadcast(broadcasthelicities);
+
+    // Set all off-diagonal elements to 1
```

Comment:
The comment says you want to set to 1.0, but then you set off diagonal elements to 0.0.
Also, you may run the loop only for j < i, and fill both cell (i, j) and (j, i) in the same scope (the matrix is so small that it won't really affect performances, though) . In this way you don't event need to check i != j